src/puptoo/app.py

```diff
@@ -223,14 +224,23 @@ def handle_message(msg, service):
                 if msg.get("custom_metadata") is None:
                     msg["custom_metadata"] = {}
                 msg["custom_metadata"]["yum_updates"] = facts["system_profile"]["yum_updates"]
+                yum_updates = msg["custom_metadata"]["yum_updates"]
+
                 # delete yum_updates from system_profile as it is already present under custom_metadata 
                 del facts["system_profile"]["yum_updates"]
 
         # Override archive hostname with name provided by client metadata
         if msg["metadata"].get("display_name"):
             facts["display_name"] = msg["metadata"]["display_name"]
         if msg["metadata"].get("ansible_host"):
-            facts["ansible_host"] = msg["metadata"]["ansible_host"]
+            facts["ansible_host"] = msg["metadata"]["ansible_host"]   
+
+        # Upload yum_updates to s3         
+        try:
+            upload_object(yum_updates, extra, msg)
+        except:
+            logger.exception("Error occurred while uploading object.")
```

Comment:
Before calling `upload_object` we should check if `yum_updates` is set. In some case we might get below result. 
```
Traceback (most recent call last):
  File "/tmp/tmp/insights-puptoo/src/puptoo/app.py", line 240, in handle_message
    upload_object(yum_updates, extra, msg)
UnboundLocalError: local variable 'yum_updates' referenced before assignment
```